### ISSUE TITLE
Could captum be used for the saliency maps of Image Segmentation model?

### ISSUE THREAD

**Issue by crissallan** [pytorch/captum#343]
Could captum be used for the saliency maps of Image Segmentation model? Such as UNet or Mask R-CNN?

**Comment by bilalsal** [comment]
Hi @crissallan, yes you should be able to use Captum to attribute neuron output to the input pixels and show the results in a "saliency map". Have a look at https://captum.ai/api/neuron.html

For example, in UNet, you can choose one neuron at layer in the encoder to find which parts of your network input it responds to. The same apply to feature-extraction neurons in Mask-RCNN.

Now, to attribute the final model output to the input pixels is a bit tricky in the case of segmentation and detection, because the output is not a score per class, but a 2D image per class. Captum is designed to attribute the scalar output back to the input, not a 2D output.
Howeve,r you can still create a wrapper of your UNet that returns torch.sum(output_for_class_i). Then, you can attribute this collective output (which is a scalar value), back to the image pixels.
This tutorial, while on classification models, might give you further inspiration on what can be done in such cases https://captum.ai/tutorials/Resnet_TorchVision_Ablation

Hope this helps

**Comment by NarineK** [comment]
Thank you @crissallan for the question and @bilalsal for the reply.
It looks like we have other segmentation related issues open. Can we consolidate the issues in one place ?

**Comment by bilalsal** [comment]
Related to #119
